qiskit/algorithms/evolvers/variational/variational_principles/variational_principle.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for a Variational Principle."""
+
+from abc import ABC, abstractmethod
+from typing import Union, List, Optional, Dict
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    CircuitQFI,
+    CircuitGradient,
+    QFI,
+    Gradient,
+    CircuitStateFn,
+    CircuitSampler,
+    OperatorBase,
+    StateFn,
+)
+from qiskit.opflow.gradients.circuit_gradients import LinComb
+from qiskit.utils import QuantumInstance
+
+
+class VariationalPrinciple(ABC):
+    """A Variational Principle class. It determines the time propagation of parameters in a
+    quantum state provided as a parametrized quantum circuit (ansatz)."""
+
+    def __init__(
+        self,
+        qfi_method: Union[str, CircuitQFI] = "lin_comb_full",
+        grad_method: Union[str, CircuitGradient] = "lin_comb",
+    ) -> None:
+        """
+        Args:
+            grad_method: The method used to compute the state gradient. Can be either
+                ``'param_shift'`` or ``'lin_comb'`` or ``'fin_diff'``.
+            qfi_method: The method used to compute the QFI. Can be either
+                ``'lin_comb_full'`` or ``'overlap_block_diag'`` or ``'overlap_diag'`` or
+                ``CircuitQFI``.
+        """
+        self._qfi_method = qfi_method
+        self.qfi = QFI(qfi_method)
+        self._grad_method = grad_method
+        if self._grad_method == "lin_comb":
+            self._grad_method = LinComb()
+        self._evolution_gradient = Gradient(self._grad_method)
+        self._qfi_gradient_callable = None
+        self._evolution_gradient_callable = None
+
+    def calc_metric_tensor(
```

Comment:
We might do it. I usually prefer to add sth to the names that hints that the method does some more work instead of just returning an object (like a getter), though.